bundles/org.openhab.binding.pulseaudio/src/main/java/org/openhab/binding/pulseaudio/internal/PulseAudioAudioSource.java

```diff
@@ -0,0 +1,246 @@
+                        break;
+                    }
+                    var sourceFormat = pulseaudioHandler.getSourceAudioFormat();
+                    if (sourceFormat == null) {
+                        throw new AudioException("Unable to get source audio format");
+                    }
+                    if (!audioFormat.isCompatible(sourceFormat)) {
+                        throw new AudioException("Incompatible audio format requested");
+                    }
+                    setIdle(true);
+                    var pipeOutput = new PipedOutputStream();
+                    registerPipe(pipeOutput);
+                    var pipeInput = new PipedInputStream(pipeOutput, 1024 * 20) {
+                        @Override
+                        public void close() throws IOException {
+                            unregisterPipe(pipeOutput);
+                            super.close();
+                        }
+                    };
+                    // get raw audio from the pulse audio socket
+                    return new PulseAudioStream(sourceFormat, pipeInput, (idle) -> {
+                        setIdle(idle);
+                        if (idle) {
+                            scheduleDisconnect();
+                        } else {
+                            // ensure pipe is writing
+                            startPipeWrite();
+                        }
+                    });
+                } catch (IOException e) {
+                    disconnect(); // disconnect force to clear connection in case of socket not cleanly shutdown
+                    if (countAttempt == 2) { // we won't retry : log and quit
+                        String port = clientSocket != null ? Integer.toString(clientSocket.getPort()) : "unknown";
+                        logger.warn(
+                                "Error while trying to get audio from pulseaudio audio source. Cannot connect to {}:{}, error: {}",
+                                pulseaudioHandler.getHost(), port, e.getMessage());
+                        setIdle(true);
+                        throw e;
+                    }
+                } catch (InterruptedException ie) {
+                    logger.info("Interrupted during source audio connection: {}", ie.getMessage());
+                    setIdle(true);
+                    throw new AudioException(ie);
+                }
+                countAttempt++;
+            }
+        } catch (IOException e) {
+            throw new AudioException(e);
+        } finally {
+            scheduleDisconnect();
+        }
+        setIdle(true);
+        throw new AudioException("Unable to create input stream");
+    }
+
+    private synchronized void registerPipe(PipedOutputStream pipeOutput) {
+        this.pipeOutputs.add(pipeOutput);
+        startPipeWrite();
+    }
+
+    private void startPipeWrite() {
+        if (pipeWriteTask == null) {
+            this.pipeWriteTask = scheduler.submit(() -> {
```

Comment:
Please do not use scheduler (and the common thread pool) for long life thread.
Please check that such thing was not done in your recent addon implementation.

Reply:
oh, no in the other plugins I didn't do this as I created a separate executor:
```
    private final ScheduledExecutorService executor = ThreadPoolManager.getScheduledPool("OH-voice-voskstt");
```
But I did it because I didn't have the common scheduler accesible there, as I didn't know this.
I will do the same here and create a new executor, thanks  for pointing it.